tests/exclude_system_edl/enc/enc.c

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <time.h>
+
+#include "exclude_system_edl_t.h"
+
+void enc_nanosleep()
+{
+    // Since nanosleep makes an ocall, we will need to include
+    // the proper ocall in our EDL file.
+    struct timespec t;
+    t.tv_sec = 0;
+    t.tv_nsec = 100;
+    nanosleep(&t, &t);
+}
+
+OE_SET_ENCLAVE_SGX(
```

Comment:
Question: Should this be using a .conf file instead?

Reply:
Good question. Is that our guidance? I think it is.

Is there an example of a test that works for both SGX and trustzone without using the `OE_SET_ENCLAVE_*` macros that I can follow as a template?

Reply:
Will defer to @CodeMonkeyLeet on what it should be doing, but I don't know of any good tests right now.   They all either use OE_SET_ENCLAVE_* for both SGX and OPTEE, or else they use .conf and don't work for OPTEE.   It's fine with me if this gets deferred to some item that updates tests as a whole to use .conf (if that's the recommendation), if there is some issue filed to do so.

Reply:
Ok so I looked a little more and while I think this is a good idea it looks like we don't actually have support for passing .conf files in creating optee enclaves. I have updated the test to use both OE_SET_ENCLAVE* macros and opened https://github.com/openenclave/openenclave/issues/2844 to track switching to using .conf files.

Reply:
> Good question. Is that our guidance? I think it is.

There's no guidance around this currently. Samples do use .conf files, but the vast majority of tests do not, except for a handful that test signing specifically. Just in terms of consistency, I'd say use the macros, particularly since .conf files will definitely not work for OP-TEE at the moment.

> Is there an example of a test that works for both SGX and trustzone without using the `OE_SET_ENCLAVE_*` macros that I can follow as a template?

Hernan has added a set of tests that also do build in OP-TEE via the macro, and it coexists with the SGX version fine (they currently result in non-overlapping definitions anyway). See any tests in tests/CMakeLists.txt under the `if (UNIX ...)` block but not `if (OE_SGX)`